src/dal/views.py

```diff
@@ -25,9 +26,16 @@ class ViewMixin(object):
 
     def dispatch(self, request, *args, **kwargs):
         """Set :py:attr:`forwarded` and :py:attr:`q`."""
-        self.forwarded = json.loads(request.GET.get('forward', '{}'))
-        if request.method == 'POST':
-            self.forwarded = json.loads(request.POST.get('forward', '{}'))
+        try:
+            self.forwarded = json.loads(request.GET.get('forward', '{}'))
+            if request.method == 'POST':
+                self.forwarded = json.loads(request.POST.get('forward', '{}'))
+
+            if not isinstance(self.forwarded, dict):
+                raise ValueError
+        except ValueError:
+            return HttpResponseBadRequest('Invalid JSON data')
```

Comment:
While you're at it, would you like to remove some SLOCS with:

    self.forwarded = getattr(request, 'POST' if request.method == 'POST' else 'GET').get('forward', '{}')

Also, it think the code would be a better read if the try block only contains the loads() call, and then check if it's a dict and return 400 otherwise: it seems like it would be easier to read and understand with two distinct return HttpResponseBadRequest() statements rather that a raise ValueError/except ValueError.